client_side/.infrastructure/setup.sh

```diff
@@ -0,0 +1,238 @@
+#!/bin/bash
+# This script takes 1 argument which is the absolute path to the user experiment
+# directory.
+
+# Checks if the user has a usable graphical display. Detects X forwarding as
+# well.
+if ! xhost &> /dev/null; then
+  echo "No display detected, please make sure that you are" \
+    "setting up the experiment in an environment with a graphical display."
+  return 1
+fi
+
+# Automatically export any assigned variables
+set -a
+
+################################################################################
+#                              CONSTANT DEFINITIONS                            #
+################################################################################
+
+# Saves the old value of PROMPT_COMMAND, since Bash Preexec overwrites it.
+PROMPT_COMMAND_OG=${PROMPT_COMMAND}
+
+# The absolute path to the user experiment directory
+EXP_DIR="$1"
+# The absolute path to the experiment's infrastructure directory
+INFRA_DIR="${EXP_DIR}/$(dirname ${BASH_SOURCE[0]})"
```

Comment:
`$BASH_SOURCE` contains [the relative
path](https://stackoverflow.com/a/35006505) of the containing script. For
example,
```sh
$ ls -R
.:
script1 script2 dir

./dir:
script3
$ cat script1
#!/bin/bash
source ./dir/script3
$ cat script2
#!/bin/bash
cd ./dir
source ./script3
$ cat dir/script3
#!/bin/bash
echo "${BASH_SOURCE[0]}"
$ ./script1
./dir/script3
$ ./script2
./script3
```

Because `configure`
[`cd`'s](https://github.com/anhnamtran/tellina_user_experiment/pull/3/files#diff-e5ddb23605f4c5338bc03c469733d118R6)
into `${EXP_DIR}` then runs
```
source .infrastructure/infrastructure.sh
```
The `${BASH_SOURCE[0]}` will always return `.infrastructure/infrastructure.sh`.